### PR TITLE
Add support for X509 client cert & key files

### PR DESCRIPTION
**Pull request by rohansingh**
As an alternative to generating an X509 certificate from the public keys in
ssh-agent, we now support supplying an X509 client certificate and key for
TLS authentication.

### CODE REVIEW

**Comment by rohansingh** [comment]
@mattnworb @gimaker @davidxia @negz

**Comment by gimaker** [comment]
:+1:

**Comment by gimaker** [comment]
Fix the build, though (checkstyle errors)

**Comment by rohansingh** [comment]
Guess I'm just not stylish enough.

**Comment by mattnworb** [comment]
Does the stuff in `CertificateFileHttpsHandler` with loading a Keystore instance load the user's keystore from disk? Why is the password hardcoded? How come we have to add something to the keystore to load the certificate?

File: helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java [hunk 1]
```diff
@@ -502,10 +505,22 @@ public static Builder newBuilder() {
...
+    private Path clientCertificatePath;
+    private Path clientKeyPath;
     private Supplier<List<Endpoint>> endpointSupplier;
     private boolean sslHostnameVerification = true;
 
+    private Builder() {
+      final String heliosCertPath = System.getenv(HELIOS_CERT_PATH);
+      if (!isNullOrEmpty(heliosCertPath)) {
+        this.clientCertificatePath = Paths.get(heliosCertPath, "cert.pem");
+        this.clientKeyPath = Paths.get(heliosCertPath, "key.pem");
```
Comment (mattnworb) [inline helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java]: are these standard names for these files? You could also just set one env var for each path.

**Comment by rohansingh** [comment]
@mattnworb Nope, it doesn't load a keystore from disk. That would be the Java(tm) way to do it, but nobody keeps their certs and keys in Java keystores so it would be a pain in the ass.

Instead, we create a keystore in memory and put the certificate and key from disk into that keystore. The keystore needs a password when we put the key into it, even though it's only going to exist for the lifetime of the process. So we just have some random password that we use. Most of this code is lifted from spotify/docker-client.

File: helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java (hunk 1 above)
Comment (rohansingh) [inline helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java]: I stuck to these since it's how `DOCKER_CERT_PATH` works.

**Comment by mattnworb** [comment]
@rohansingh ah cool. I think you could stick that whole second paragraph in the source as comments and it'd be :ok_hand:

File: helios-client/src/main/java/com/spotify/helios/client/AuthenticatingHttpConnector.java [hunk 2]
```diff
@@ -63,19 +66,31 @@
...
 
   @VisibleForTesting
   AuthenticatingHttpConnector(final String user,
                               final Optional<AgentProxy> agentProxyOpt,
+                              final Optional<Path> clientCertificatePath,
+                              final Optional<Path> clientKeyPath,
                               final EndpointIterator endpointIterator,
                               final DefaultHttpConnector delegate,
                               final List<Identity> identities) {
+    if (clientCertificatePath.isPresent() ^ clientKeyPath.isPresent()) {
```
Comment (mattnworb) [inline helios-client/src/main/java/com/spotify/helios/client/AuthenticatingHttpConnector.java]: u fancy

**Pull request merged**